semgrep-core/bin/Main.ml

```diff
@@ -415,18 +421,10 @@ let iter_generic_ast_of_files_and_get_matches_and_exn_to_errors f files =
        if !verbose then pr2 (spf "Analyzing %s" file);
        try
          let lang =
-           match Lang.langs_of_filename file with
-           | [lang] -> lang
-           | x::_xs ->
-               (match Lang.lang_of_string_opt !lang with
-               | Some lang -> lang
-               | None ->
-                 pr2 (spf "no language specified, defaulting to %s for %s"
-                      (Lang.string_of_lang x) file);
```

Comment:
I think you want to move the `langs_of_filename` check in here, and keep the automated detection. So if someone omits `-lang` and passes in `foo.py` we parse as python, but if someone passes `foo.blargh` we error.

Reply:
I think it's fine to not work when -lang is not passed?

Reply:
OK, but then I think you'll want to audit all the documentation to make sure we have `-lang` in all our examples.

IMO it's easier to make the smaller change here, but will leave it to you.